data/strings/strings.txt

```diff
@@ -2409,46 +2409,48 @@
   [maps_storage_downloaded]
     comment = E.g. "Downloaded maps: 500Mb" in Maps Storage settings
     tags = android
-    en = Downloaded maps
-    af = Afgelaaide kaarte
-    ar = الخرائط المنزلة
-    az = Xəritələr
-    be = Спампаваныя мапы
-    bg = Карти
-    ca = Mapes baixats
-    cs = Mapy
-    da = Kort
-    de = Heruntergeladene Karten
-    el = Κατεβασμένοι Χάρτες
-    es = Mapas descargados
-    et = Kaardid
-    eu = Deskargatutako mapak
-    fa = نقشه‌ها
-    fi = Ladatut kartat
-    fr = Cartes téléchargées
-    he = מפות שהורדו
-    hi = मानचित्र डाउनलोड किए गए
-    hu = Letöltött térképek
-    id = Peta
-    it = Mappe
-    ja = マップ
-    ko = 지도
-    lt = Atsisiųsti žemėlapiai
-    nb = Kart
-    nl = Kaarten
-    pl = Mapy
-    pt = Mapas
-    pt-BR = Mapas
-    ro = Hărți
-    ru = Загруженные карты
-    sk = Mapy
-    sv = Kartor
-    th = แผนที่
-    tr = Haritalar
-    uk = Завантаженнi мапи
-    vi = Bản đồ
-    zh-Hans = 地图
-    zh-Hant = 地圖
+    en = Downloaded maps: %@
+    af = Afgelaaide kaarte: %@
+    ar = الخرائط التي تم تنزيلها: %@
```

Comment:
I don't know where to look in the app to answer this, but let me give you a quick crash course.

Bidirectional text is probably pretty hard to get a grasp on if you don't interact with it much (i.e. if you speak a LTR language and never/rarely encounter mixed text). But I think I can explain it in simple terms, especially to a programmer.

First and foremost: **strings (as data, or in memory) are sequences of characters, and they don't give a crap about text directionality.** So the string "a, b" is a sequence: a, then comma, then space, then b. The same is true for mixed directionality text: the string "aאבb" is the sequence: a, then א, then ב,then b. **Note that the order of the sequence doesn't match the order it appears on screen, neither LTR nor RTL.** The order of the sequence is the order in which the user should read the string.

Directionality of text only matters for rendering. Before a string gets rendered, it doesn't make sense to talk about "this character (or substring) is to the left of that character". It only makes sense to say "this character is *before* that character".

You should count yourself lucky that you don't need to implement the rendering logic. However, you do sometimes need to give the renderer hints about the directionality of text. When a string gets rendered it usually has a directionality, and if you don't specify it, the renderer will use some heuristic to figure it out. An extremely common heuristic is: the first strongly-directional character in the string will determine the directionality of the rest of the string. So consider the following sentence:

Organic Maps היא אפליקציית מפות מעולה. אני נותן לה ציון A+!

(translation: Organic Maps is an excellent maps app. I give it a score of A+!)

The sentence starts with the Latin letter O, so that decides the direction of the entire sentence. **However, this is bad, as this is a Hebrew sentence that just has some Latin text mixed in.** It's very difficult for a reader to read If we instead start with a Hebrew letter:

האפליקצייה Organic Maps היא אפליקציית מפות מעולה. אני נותן לה ציון A+!

(translation: The app Organic Maps is an excellent maps app. I give it a score of A+!)

Now the first character is a Hebrew letter, and the whole string gets rendered right-to-left as it's supposed to. **Note that the string is exactly the same, with only characters prepended before the sequence.** The visual changes are all in the rendering.

So how is this relevant here? Since in the `ar` string above the first character is Arabic, come render time the operating system should render the entire string right-to-left. The colon and variable appear **later** in the string, so on the screen they should appear to the left.

The way it's rendered here, or in your editor, or wherever else is not really important. What's important is the order of the characters and the way it's rendered in the actual app. Also note that some editors (not all, and IDK about github's web interface) use this first character heuristic logic on a per-line basis, but because each line starts with `ar` it will be LTR anyway.

Reply:
> I don't know where to look in the app to answer this, but let me give you a quick crash course.
> 
You can find your changes in debug app at "settings/Save maps to"

> In the example sentence above, the "A+" was rendered with the + to the left of the A, which is not the desired outcome. This is because + has no directionality so it inherits the RTL context. I could use a [left-to-right mark](https://en.wikipedia.org/wiki/Left-to-right_mark) to make it appear correct:
> 
As much as I understand you tried to insert the LTR text (eg. 24 MB) in RTL language, without affecting  LTR's actual direction. But i guess this thing already exist in OM. I am not sure about special characters but for english inside any RTL language it comes in proper form. Below is the screenshot

![1714165388515](https://github.com/organicmaps/organicmaps/assets/120750626/ffff4ed2-7fcb-4c92-97fc-592972e72362)

I Locally ran your changes in my android and i saw no changes in before and after. Below are some screenshots of comparisons.

ARABIC:-

[Before]
![ar_before](https://github.com/organicmaps/organicmaps/assets/120750626/a66409ef-a81f-40fd-a78f-04f77bafa1b1)
[After]
![ar_after](https://github.com/organicmaps/organicmaps/assets/120750626/95983383-0082-4d26-bbcc-abb28af5707b)

PERSIAN:-

[Before]
![per_before_r](https://github.com/organicmaps/organicmaps/assets/120750626/cbf00884-6e6a-4d5d-8f09-b0281d2dfaf4)
[After]
![per_after](https://github.com/organicmaps/organicmaps/assets/120750626/ebb4ebce-fb2d-46c0-b9bc-a62104a4bd48)

Reply:
> As much as I understand you tried to insert the LTR text (eg. 24 MB) in RTL language, without affecting LTR's actual direction. But i guess this thing already exist in OM. I am not sure about special characters but for english inside any RTL language it comes in proper form. Below is the screenshot

Yeah, it was a bit of a tangent. The points I was trying to make are: 

- The order of the characters in the string is what matters, and it is seemingly correct
- The text in the app is usually rendered by the operating system, which does a dang good job of rendering things correctly almost all of the time
- The way it gets rendered anywhere else other than inside OM doesn't matter - e.g. in strings.txt on GitHub, in code editors, or wherever else
- There are edge cases that require special care but they are rare and you shouldn't worry about them. The case of this PR isn't one of them.

> [screenshots]

I can't read either language but I recognise the numeric digits (٧٠) and the colon, this seems correct.

Reply:
@rtsisyk Should be ok